src/components/atoms/TextField/index.tsx

```diff
@@ -0,0 +1,199 @@
+  purchaseDate: string;
+}
+
+const returnOutlinedtextFieldType = (textFieldType: TextFieldType) => {
+  switch (textFieldType) {
+    case 'email':
+      return 'email';
+    case 'purchaseDate':
+      return 'date';
+    default:
+      return 'text';
+  }
+};
+
+const returnOutlinedInputValue = (
+  value: TextField,
+  textFieldType: TextFieldType
+) => {
+  switch (textFieldType) {
+    case 'email':
+      return value.email;
+    case 'password':
+      return value.password;
+    case 'confirmPassword':
+      return value.confirmPassword;
+    case 'bookName':
+      return value.bookName;
+    case 'bookPrice':
+      return value.bookPrice;
+    case 'purchaseDate':
+      return value.purchaseDate;
+  }
+};
+
+const returnInputLabel = (textFieldType: TextFieldType) => {
+  switch (textFieldType) {
+    case 'email':
+      return 'メールアドレス';
+    case 'password':
+      return 'パスワード';
+    case 'confirmPassword':
+      return 'もう一度確認';
+    case 'bookName':
+      return '書籍名';
+    case 'bookPrice':
+      return '値段';
+    case 'purchaseDate':
+      return '';
+  }
+};
+
+const returnTextFieldLabelWidth = (textFieldType: TextFieldType) => {
+  switch (textFieldType) {
+    case 'email':
+      return 110;
+    case 'password':
+      return 80;
+    case 'confirmPassword':
+      return 100;
+    case 'bookName':
+      return 50;
+    case 'bookPrice':
+      return 40;
+    case 'purchaseDate':
+      return 0;
+  }
+};
+
+const checktextFieldType = (textFieldType: TextFieldType) => {
+  return textFieldType == 'password' || textFieldType == 'confirmPassword';
+};
+
+export default function TextField(props: Props) {
+  const classes = useStyles();
+  const { textFieldType } = props;
+  const [values, setValues] = React.useState<State>({
+    email: '',
+    password: '',
+    confirmPassword: '',
+    bookName: '',
+    bookPrice: 0,
+    purchaseDate: '',
+    mailValidate: true,
+    passwordValidate: true,
+    showIcon: false,
+    showPassword: checktextFieldType(textFieldType),
+  });
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleClickShowPassword = () => {
+    setValues({ ...values, showPassword: !values.showPassword });
+  };
+
+  const handleChange = (prop: keyof State) => (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setValues({ ...values, [prop]: event.target.value });
+  };
+
+  const handleMouseDownPassword = (
+    event: React.MouseEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+  };
+
+  const renderPasswordIcon = () => {
+    return (
+      <InputAdornment position="end">
+        <IconButton
+          aria-label="toggle password visibility"
+          onClick={handleClickShowPassword}
+          onMouseDown={handleMouseDownPassword}
+          edge="end"
+        >
+          {values.showPassword ? <VisilityIcon /> : <VisilityIconOff />}
+        </IconButton>
+      </InputAdornment>
+    );
+  };
+
+  const validate = () => {
```

Comment:
`is~~`

Reply:
https://github.com/khayashi-cq/BookManager-UserScreen/pull/1#discussion_r635043715